bench/bulk_update/main.c

```diff
@@ -98,11 +99,20 @@ callback_done(void * cookie, int failed)
 	if ((tv_now.tv_sec > C->tv_end.tv_sec) ||
 	    ((tv_now.tv_sec == C->tv_end.tv_sec) &&
 		(tv_now.tv_usec > C->tv_end.tv_usec))) {
-		C->done = 1;
+		/* Stop again in 1 second. */
+		C->tv_end.tv_sec = tv_now.tv_sec + 1;
+		C->tv_end.tv_usec = tv_now.tv_usec;
+
+		/* Record number in a new array position. */
+		C->Npos++;
+
+		/* Should we stop now? */
+		if (C->Npos >= BENCHMARK_SECONDS)
+			C->done = 1;
 	} else if ((tv_now.tv_sec > C->tv_start.tv_sec) ||
 	    ((tv_now.tv_sec == C->tv_start.tv_sec) &&
 		(tv_now.tv_usec > C->tv_start.tv_usec))) {
-		C->N += 1;
+		C->Ns[C->Npos] += 1;
```

Comment:
Looks like you're not counting the first completion from each of the 2nd through 10th seconds?